src/app/search-bar/search-bar.component.ts

```diff
@@ -1,22 +1,49 @@
-import { Component, OnInit } from '@angular/core';
+import { Component } from '@angular/core';
 import { NgForm } from '@angular/forms';
 
 import { minInput } from '@core/models/matching.models';
+import { VideoServices } from '@core/models/videoServices.model';
+import { VimeoService as HttpService } from '@core/services/http.service';
 import { InputService } from '@core/services/input.service';
+import { Video } from '@core/models/video.model';
+import { VimeoResponse, YouTubeResponse } from '@core/models/video-response.model';
+import { NewVideoService } from '@core/services/new-video.service';
+import { VideosListService } from '@core/services/videos-list.service';
 
 @Component({
-  selector: 'app-search-bar',
-  templateUrl: './search-bar.component.html',
-  styleUrls: ['./search-bar.component.scss']
+	selector: 'app-search-bar',
+	templateUrl: './search-bar.component.html',
+	styleUrls: ['./search-bar.component.scss'],
 })
 export class SearchBarComponent {
-  public isValidationErrorVisible : boolean
-  public minInput = minInput
-  
-  constructor(private inputService: InputService) {}
+	public isValidationErrorVisible: boolean;
+	public minInput = minInput;
 
-  public onAddVideo(form: NgForm) {
-    if (form.invalid) this.isValidationErrorVisible  = true
-    const videoData = this.inputService.onProviderAndIdCheck(form.value.value)
-  }
+	constructor(
+		public inputService: InputService,
+		public httpService: HttpService,
+		public newVideoService: NewVideoService,
+		public videoListService: VideosListService,
+	) {}
+
+	public onAddVideo(form: NgForm): void {
+		let newVideo: Video;
+		if (form.invalid) this.isValidationErrorVisible = true;
+		const inputData = this.inputService.onProviderAndIdCheck(form.value.value);
+
+		if (inputData.provider === VideoServices.vimeo) {
+			this.httpService.getVimeoData(inputData.id).subscribe((date: VimeoResponse) => {
```

Comment:
tutaj brakuje unsubscribe

Reply:
przy http nie trzeba podobni samo się unsubscrybuje

Reply:
Generalnie tak ale zawsze można dodać aby weszło w krew bo to naprawdę może powodować sporo problemy. No i trochę mi się to nie podoba, bo patrz w konstruktorze masz kilka serwisów i wszystkie robią coś z video, może warto by to przebudować tak aby za zarządzanie video odpowiadał jeden serwis. Aktualnie w komponencie masz też logikę która odpowiada za tworzenie nowego video itd. Ten komponent powinien tylko przyjmować jakieś dane od usera, validować  i przekazywać je dalej, za dużo tu logiki. Jeśli masz już serwis to nie warto pchać takiej logiki do komponentu.